### DISCUSSION TITLE
Does Swoole Client replaces gRPC ?

### DISCUSSION THREAD

**Discussion by fakharak** [swoole/swoole-src#4218] [Ideas]
I am asking this question because i want my Network solution to rely less on third-party protocols like gRPC. I am exploring this idea that "If i can use Swoole Client (or, Swoole's Courtine Client) from within Swoole Server (or Swoole Process) then it means my Swoole server components do not need gRPC to communicate with each other, because Swoole Client supports TCP, UDP, HTTP1/2 (with SSL), MQTT, FAST-CGI (in addition to gRPC which itself uses HTTP). 
Am i getting it correct ?

**Comment by sy-records** [comment]
yeah, do it the way you want.

**Reply by fakharak** [reply]
Thanks, that helps in Design Patterns with Swoole.
